fin_control/src/fin_control.cpp

```diff
@@ -193,46 +193,40 @@ void FinControl::reportAngles()
 {
   const char *log;
   fin_control::ReportAngle message;
-
-  if (!reportAnglesEnabled) return;  // do not interfere with fins updating
+  bool servos_angles_info_complete = true;
 
   message.header.stamp = ros::Time::now();
-  message.angle_in_radians = 0;
-
-  boost::mutex::scoped_lock lock(m_mutex);
-
-  for (int x = 0; x < numOfIDs; x++)
-  {
-    reportAngle(ids[x]);
-  }
-}
-void FinControl::reportAngle(uint8_t id)
-{
-  const char *log;
-  fin_control::ReportAngle message;
+  for (int i = 0; i < numOfIDs; i++) message.angles_in_radians.push_back(0.);
 
   if (!reportAnglesEnabled) return;  // do not interfere with fins updating
 
-  message.header.stamp = ros::Time::now();
-  message.angle_in_radians = 0;
-
-  message.ID = id;
-  // get from dynamixel
-  if (myWorkBench.getRadian(id, &message.angle_in_radians, &log))
+  for (int id = 0; id < numOfIDs; id++)
   {
-    message.angle_in_radians = static_cast<float>(
-        round(((message.angle_in_radians / ctrlFinScaleFactor) - degreesToRadians(ctrlFinOffset)) *
-              100.0) /
-        100.0);
-    finAngleCheck.test(message.angle_in_radians);
-
-    publisher_reportAngle.publish(message);
+    // get from dynamixel
+    if (myWorkBench.getRadian(id, &message.angles_in_radians[id], &log))
+    {
+      message.angles_in_radians[id] =
+          static_cast<float>(round(((message.angles_in_radians[id] / ctrlFinScaleFactor) -
```

Comment:
@patapufate FYI `float` appears redundant

Reply:
I put it because message.angles_in_radians is float but ctrlFinScaleFactor and CtrlFinOffset are doubles.